src/storage/buffer_manager.cpp

```diff
@@ -424,6 +424,16 @@ void BufferManager::AddToEvictionQueue(shared_ptr<BlockHandle> &handle) {
 	queue->q.enqueue(BufferEvictionNode(weak_ptr<BlockHandle>(handle), handle->eviction_timestamp));
 }
 
+void BufferManager::VerifyZeroReaders(shared_ptr<BlockHandle> &handle) {
+#ifdef DEBUG
```

Comment:
Rather than doing this in debug mode, could we perhaps add a specific define for this that we enable on a dedicated CI run, similar to the `No String Inline` CI run? Always doing this in debug mode will slow debug mode down significantly - plus we might be missing several bugs because debug tests are always run on the fast tests. I think we might also want to run this on all tests in release mode.

Reply:
Good idea. Am testing this locally, and this seems to have found a bug in `test/sql/constraints/foreignkey/test_fk_tpch.test_slow` - My guess is that `PhysicalIndexJoin`has an issue too.

Reply:
I've fixed the bug and created a new CI run with this `ifdef`.

Reply:
Excellent! Looks great :)